concepts/05 UI Components/DataGrid/20 Editing/43 Customize Edit Form/10 Appearance and Editing/10 Form Layout Customization.md

```diff
@@ -0,0 +1,189 @@
+                    editor-type = "dxTextArea"
+                    :editor-options = "columnEditorOptions"
+                >
+                    <DxLabel location="top" />
+                </DxFormItem>
+            </DxColumn>
+        </DxDataGrid>
+    </template>
+
+    <script>
+    import 'devextreme/dist/css/dx.light.css';
+
+    import DxDataGrid, {
+        DxEditing,
+        DxColumn,
+        DxFormItem,
+        DxLabel
+    } from 'devextreme-vue/data-grid';
+
+    export default {
+        components: {
+            DxDataGrid,
+            DxEditing,
+            DxColumn,
+            DxFormItem,
+            DxLabel
+        },
+        data() {
+            return {
+                columnEditorOptions: { height: 100 }
+            }
+        }
+        // ...
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import DataGrid, {
+        Editing,
+        Column,
+        FormItem,
+        Label
+    } from 'devextreme-react/data-grid';
+
+    const defaultEditorOptions = { height: 100 }
+
+    export default function App() {
+        return (
+            <DataGrid ... >
+                <Editing
+                    allowUpdating={true}
+                    mode="form"
+                />
+                <Column dataField="ID">
+                    <FormItem visible={false} />
+                </Column>
+                <Column dataField="Notes">
+                    <FormItem 
+                        colSpan={2}
+                        cssClass = "custom-class"
+                        editorType = "dxTextArea"
+                        editorOptions = "{ defaultEditorOptions }"
+                    >
+                        <Label location="top" />
+                    </FormItem>
+                </Column>
+            </DataGrid>
+        );
+    }
+
+---
+
+If you need to implement major changes (for example, arrange items and groups), use the **editing**.[form](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/#form) property. Declare the required items for this object and specify the [dataField](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#dataField) property. 
+
+DataGrid automatically generates editors with the related event handlers for the form items based on column settings and other DataGrid options like [onEditorPrepared](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#dataField). Do not use the [editCellTemplate](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#editCellTemplate) or [editorOptions](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#editorOptions) properties for items that are bound to **dataField** in the **editing.form** configuration. You can add unbound items just like for a regular [Form](/Documentation/ApiReference/UI_Components/dxForm/) component.
```

Comment:
-> Add unbound items where? I cannot suggest a change because I do not have a final vision of this paragraph.

Look at this section https://js.devexpress.com/Documentation/Guide/UI_Components/Form/Configure_Simple_Items/#Create_an_Unbound_Simple_Item

Reply:
I missed this link in the text above...

Reply:
I see, you updated this one. I'll mark this thread as resolved.